File: backend/src/main/java/com/medcollect/api/controllers/UserController.java
package com.medcollect.api.controllers;

import com.medcollect.api.dtos.CreateUserRequest;
import com.medcollect.api.models.User;
import com.medcollect.api.services.UserService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

import java.util.Optional;

@RestController
@RequestMapping("/api/users")
@RequiredArgsConstructor
public class UserController {
    private final UserService userService;

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public User createUser(@RequestBody @Valid CreateUserRequest request) {
        return userService.createUser(request);
    }

    @GetMapping("/firebase/{firebaseUid}")
    public Optional<User> getUserByFirebaseUid(@PathVariable String firebaseUid) {
        return userService.getUserByFirebaseUid(firebaseUid);
    }
}

File: backend/src/main/kotlin/com/medcollect/api/controllers/UserController.kt
package com.medcollect.api.controllers

import com.medcollect.api.dtos.CreateUserRequest
import com.medcollect.api.models.User
import com.medcollect.api.services.UserService
import jakarta.validation.Valid
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/users")
class UserController(
    private val userService: UserService
) {
    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    fun createUser(@RequestBody @Valid request: CreateUserRequest): User =
        userService.createUser(request)

    @GetMapping("/firebase/{firebaseUid}")
    fun getUserByFirebaseUid(@PathVariable firebaseUid: String): User? =
        userService.getUserByFirebaseUid(firebaseUid)
}
